### DISCUSSION TITLE
[Help] Error after update tp neovim 0.10.0

### DISCUSSION THREAD

**Discussion by devmkdd** [zk-org/zk-nvim#186] [Help] [answered]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

I've just updated neovim from prio 0.10 to 0.10.1 and currently I can't zk-nvim to work as every command within neovim leads to the following error:

`Error executing Lua callback: .../.local/share/nvim/lazy/zk-nvim/lua/zk/api.lua:19: attempt to index a nil value
stack traceback:
	.../.local/share/nvim/lazy/zk-nvim/lua/zk/api.lua:19: in function 'execute_command'
	.../.local/share/nvim/lazy/zk-nvim/lua/zk/api.lua:33: in function 'index'
	.../.local/share/nvim/lazy/zk-nvim/lua/zk.lua:87: in function 'fn'
	.../.local/share/nvim/lazy/zk-nvim/lua/zk/commands/init.lua:19: in function <.../.local/share/nvim/lazy/zk-nvim/lua/zk/commands/init.lua:12>
`

I've also updated to the latest commit as I thought [this commit](https://github.com/zk-org/zk-nvim/commit/1f331d994d90f1dc93d023508951be437d19813e#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) would address the problem.

But it does. Currently I'm running 
zk-nvim on [dbf4eeab55b08856c9d6b6722dbff39630bb35eb](https://github.com/zk-org/zk-nvim/commit/dbf4eeab55b08856c9d6b6722dbff39630bb35eb)
zk 0.14.0
neovim 0.10.1

and this is my config using lazy.nvim

```
return {
  "zk-org/zk-nvim",
  config = function()
		require("zk").setup({
	--		picker = "telescope",
			lsp = {

				config = {
					cmd  = { "zk" },
					name = { "zk" }

				},
				-- automatically attach buffers in a zk notebkook that match the given filetypes
				auto_attach = {
					enabled = true,
					filetypes = { "markdown" },
				},
			},
		})
	end,
}
```

Did I miss something for the transition?

Thanks for your help!

**Comment by tjex** [comment, marked as answer]
Nothing needs to be changed config wise for the new version.

But your current config is incorrect to begin with. The `cmd` is `zk lsp`.

See the [setup section](https://github.com/zk-org/zk-nvim#setup), which shows the correct config.

It's also worth fully closing any neovim session that's running in a zk notebook after updating, and then opening fresh. Just in case something plays weird.

Let me know if this fixes the issue!

```lua
require("zk").setup({
  -- can be "telescope", "fzf", "fzf_lua", "minipick", or "select" (`vim.ui.select`)
  -- it's recommended to use "telescope", "fzf", "fzf_lua", or "minipick"
  picker = "select",

  lsp = {
    -- `config` is passed to `vim.lsp.start_client(config)`
    config = {
      cmd = { "zk", "lsp" },
      name = "zk",
      -- on_attach = ...
      -- etc, see `:h vim.lsp.start_client()`
    },

    -- automatically attach buffers in a zk notebook that match the given filetypes
    auto_attach = {
      enabled = true,
      filetypes = { "markdown" },
    },
  },
})
```

**Reply by devmkdd** [reply]
This seems to work, thank you!
